PerimeterXModule/Internals/CredentialsIntelligence/LoginSuccessful/AdditionalS2SUtils.cs

```diff
@@ -0,0 +1,53 @@
+﻿namespace PerimeterX
+{
+    public class AdditionalS2SUtils
+    {
+        public static Activity CreateAdditionalS2SActivity(PxModuleConfigurationSection config, int? statusCode, bool? loginSuccessful, PxContext pxContext)
+        {
+            bool isBreachedAccount = pxContext.IsBreachedAccount();
+            LoginCredentialsFields loginCredentialsFields = pxContext.LoginCredentialsFields;
+
+            bool shouldAddRawUsername = isBreachedAccount &&
+                config.SendRawUsernameOnAdditionalS2SActivity &&
+                loginCredentialsFields.RawUsername != null;
+
+            AdditionalS2SActivityDetails details = new AdditionalS2SActivityDetails
+            {
+                ModuleVersion = PxConstants.MODULE_VERSION,
+                ClientUuid = pxContext.UUID,
+                RequestId = pxContext.RequestId,
+                CiVersion = config.CiVersion,
+                CredentialsCompromised = isBreachedAccount,
+                RawUsername = shouldAddRawUsername ? loginCredentialsFields.RawUsername : null,
+                SsoStep = config.CiVersion == "multistep_sso" ? loginCredentialsFields.SsoStep : null
+            };
+
+            if (statusCode.HasValue)
+            {
+                details.HttpStatusCode = statusCode.Value;
+            }
+
+            if (loginSuccessful.HasValue)
+            {
+                details.LoginSuccessful = loginSuccessful.Value;
```

Comment:
If this has a value, we should re-evaluate here if the raw username should be sent. It should only be sent if the following three conditions are met:
(1) Enabled in the configuration
(2) Credentials are compromised
(3) Login is successful

We will always know (1) and (2) when we create the additional_s2s activity, but (3) may or may not be known. If it is known here, then let's remove the raw username if it the login was not successful.

Reply:
DONE